src/lib/components/ui/ScrollView/ScrollView.jsx

```diff
@@ -0,0 +1,256 @@
+
+  const scrollPositionStart = direction === 'horizontal' ? 'left' : 'top';
+  const scrollPositionEnd = direction === 'horizontal' ? 'right' : 'bottom';
+  const scrollViewContentEl = useRef(null);
+  const scrollViewViewportEl = useRef(null);
+
+  const handleScrollViewState = (currentPosition) => {
+    const isScrolledAtStartActive = currentPosition[scrollPositionStart] < 0;
+    const isScrolledAtEndActive = currentPosition[scrollPositionEnd] > 0;
+
+    if (isScrolledAtStartActive !== isScrolledAtStart) {
+      setIsScrolledAtStart(isScrolledAtStartActive);
+    }
+
+    if (isScrolledAtEndActive !== isScrolledAtEnd) {
+      setIsScrolledAtEnd(isScrolledAtEndActive);
+    }
+  };
+
+  useEffect(
+    () => {
+      handleScrollViewState(
+        getElementsPositionDifference(scrollViewContentEl, scrollViewViewportEl),
+      );
+    },
+    [], // eslint-disable-line react-hooks/exhaustive-deps
+  );
+
+  useLoadResize(
+    (currentPosition) => (handleScrollViewState(currentPosition)),
+    [isScrolledAtStart, isScrolledAtEnd],
+    scrollViewContentEl,
+    scrollViewViewportEl,
+    debounce,
+  );
+
+  useScrollPosition(
+    (currentPosition) => (handleScrollViewState(currentPosition)),
+    [isScrolledAtStart, isScrolledAtEnd],
+    scrollViewContentEl,
+    scrollViewViewportEl,
+    debounce,
+  );
+
+  const setAlpha = (rgba, alpha) => {
+    const newColor = { ...rgba };
+    newColor.alpha = alpha;
+    return newColor;
+  };
+
+  const rgbaToString = (rgba) => (
+    `rgba(${rgba.red}, ${rgba.green}, ${rgba.blue}, ${rgba.alpha})`
+  );
+
+  const inlineStyle = (
+    scrollDirection,
+    customArrowsColor,
+    customShadowColor,
+    customShadowSize,
+    startShadowStyle,
+    endShadowStyle,
+  ) => {
+    const shadowStartColor = rgbaToString(customShadowColor);
+    const shadowEndColor = rgbaToString(setAlpha(customShadowColor, 0));
+
+    /* eslint-disable sort-keys */
+    return {
+      '--rui-local-arrow-color': customArrowsColor || 'inherit',
+      '--rui-local-shadow-width': scrollDirection === 'horizontal' ? shadowSize : 'auto',
+      '--rui-local-shadow-height': scrollDirection === 'horizontal' ? 'auto' : shadowSize,
+      '--rui-local-start-shadow-background':
+        startShadowStyle.background
+        || `linear-gradient(${direction === 'horizontal' ? 'to right' : 'to bottom'}, ${shadowStartColor}, ${shadowEndColor})`,
+      '--rui-local-start-shadow-box-shadow': startShadowStyle.boxShadow || 'none',
+      '--rui-local-end-shadow-background':
+        endShadowStyle.background
+        || `linear-gradient(${direction === 'horizontal' ? 'to left' : 'to top'}, ${shadowStartColor}, ${shadowEndColor})`,
+      '--rui-local-end-shadow-box-shadow': endShadowStyle.boxShadow || 'none',
+    };
+    /* eslint-enable sort-keys */
+  };
+
+  const arrowHandler = (contentEl, viewportEl, scrollViewDirection, shiftDirection, step) => {
+    const differenceX = () => {
```

Comment:
It is more efficient not to wrap calculations like this into functions. `differenceX` and `differenceY` are used only here, so it is better to write somethink like:

```
let differenceX = 0;
let differenceY = 0;

if (scrollViewDirection === 'horizontal') {
         differenceX = shiftDirection === 'next'
           ? step
           : -1 * step;
}

// ...

```

Reply:
Also the code is the same for `differenceX` and `differenceY`. i think it could be abstracted to a function.

Reply:
I like it this way, 10 lines shorter now. Is that OK?

```js
  const arrowHandler = (contentEl, viewportEl, scrollViewDirection, shiftDirection, step) => {
    let differenceX = 0;
    let differenceY = 0;

    if (scrollViewDirection === 'horizontal') {
      differenceX = shiftDirection === 'next' ? step : -1 * step;
    } else {
      differenceY = shiftDirection === 'next' ? step : -1 * step;
    }

    viewportEl.current.scrollBy(differenceX, differenceY);
  };
```

Reply:
How about:
```javascript
const arrowHandler = (contentEl, viewportEl, scrollViewDirection, shiftDirection, step) => {
    const offset = shiftDirection === 'next' ? step : -1 * step;
    const differenceX = scrollViewDirection === 'horizontal' ? offset : 0;
    const differenceY = scrollViewDirection !== 'horizontal' ? offset : 0;

    viewportEl.current.scrollBy(differenceX, differenceY);
  };
```

Reply:
From 22 lines to 6, not bad!

Reply:
We could shorten it to something like:

```javascript
   viewportEl.current.scrollBy(['horizontal', 'vertical'].map((dir) => scrollViewDirection === dir ? (shiftDirection === 'next' ? step : -1 * step) : 0);
```
but IMHO it ios not as reaadable